args.c

```diff
@@ -24,8 +24,8 @@ void parse_args(int argc, char* argv[])
 	// Regardless of any arguments, we always pass "-k" (nofork) to dnsmasq
 	argc_dnsmasq = 2;
 	argv_dnsmasq = calloc(argc_dnsmasq, sizeof(char*));
-	argv_dnsmasq[0] = "";
-	argv_dnsmasq[1] = "-k";
+	argv_dnsmasq[0] = (char*)"";
+	argv_dnsmasq[1] = (char*)"-k";
```

Comment:
Why does this cast need to exist? What other type could a string literal be?

Reply:
`""` is `const char[1]`. However, `argv_dnsmasq` is `char*`.

```c++
argv_dnsmasq[0] = "";
```
results in the compilation error
```
error: assignment discards ‘const’ qualifier from pointer target type
```

Reply:
Can we change `argv_dnsmasq` to `const char**` then?